model/task/task.go

```diff
@@ -424,12 +424,17 @@ func (t *Task) DependenciesMet(depCaches map[string]Task) (bool, error) {
 		return true, nil
 	}
 
-	deps, err := t.populateDependencyTaskCache(depCaches)
+	_, err := t.populateDependencyTaskCache(depCaches)
 	if err != nil {
 		return false, errors.WithStack(err)
 	}
 
-	for _, depTask := range deps {
+	for _, dependency := range t.DependsOn {
+		depTask, ok := depCaches[dependency.TaskId]
+		// ignore non-existent dependencies
+		if !ok {
+			continue
```

Comment:
Should we log, since something has gone wrong?

Reply:
done